lnrpc/routerrpc/forward_interceptor.go

```diff
@@ -119,6 +121,37 @@ func (r *forwardInterceptor) resolveFromClient(
 			Action: htlcswitch.FwdActionResume,
 		})
 
+	case ResolveHoldForwardAction_RESUME_MODIFIED:
+		// Modify HTLC and resume forward.
+		outgoingAmtMsat := fn.None[lnwire.MilliSatoshi]()
+		if in.OutgoingAmountMsat > 0 {
+			outgoingAmtMsat = fn.Some[lnwire.MilliSatoshi](
+				lnwire.MilliSatoshi(in.OutgoingAmountMsat),
+			)
+		}
+
+		customRecords := fn.None[record.CustomSet]()
+		if in.CustomRecords != nil && len(in.CustomRecords) > 0 {
```

Comment:
nit: don't need a nil check if we're checking for length (since `len(nil) == 0`).